pkg/distributor/query.go

```diff
@@ -184,6 +211,81 @@ func (d *Distributor) queryIngesters(ctx context.Context, replicationSet ring.Re
 	return result, nil
 }
 
+// mergeExemplarSets merges and dedupes two sets of already sorted sample pairs.
+// Defined here instead of pkg/util to avoid a import cycle.
+func mergeExemplarSets(a, b []cortexpb.Exemplar) []cortexpb.Exemplar {
+	result := make([]cortexpb.Exemplar, 0, len(a)+len(b))
+	i, j := 0, 0
+	for i < len(a) && j < len(b) {
+		if a[i].TimestampMs < b[j].TimestampMs {
+			result = append(result, a[i])
+			i++
+		} else if a[i].TimestampMs > b[j].TimestampMs {
+			result = append(result, b[j])
+			j++
+		} else {
+			result = append(result, a[i])
```

Comment:
I'm not sure, but should this compare `Value` as well?   The Prometheus dedupe logic does so I think different exemplars with the same Ts but different Values are possible. https://github.com/prometheus/prometheus/blob/main/pkg/exemplar/exemplar.go#L49

Reply:
Thanks again for the detailed explanation.